consensus/istanbul/backend/announce.go

```diff
@@ -937,7 +937,7 @@ func (vc *versionCertificate) payloadToSign() ([]byte, error) {
 func (sb *Backend) generateVersionCertificate(version uint) (*versionCertificate, error) {
 	vc := &versionCertificate{
 		Address:   sb.Address(),
-		PublicKey: sb.publicKey,
+		PublicKey: sb.authorizeInfo.Load().(*AuthorizeInfo).PublicKey,
```

Comment:
I think it would be nice to have a function that encapsulates the loading of the atomic value.
```suggestion
		PublicKey: sb.auth().PublicKey,
```

Reply:
I actually had one but ended up deleting it because I thought it was an overkill. If you think it looks better I'll re-add it.

Reply:
I think it makes sense because you have 10 instances of this loading code, i think its nice to encapsulate the specifics of that.